.github/workflows/application-ci.yaml

```diff
@@ -23,6 +23,35 @@ on:
   workflow_dispatch:
 
 jobs:
+  # Check if there are changes related to migrations (This is used to decide whether to build migrations container build or not)
+  detect-changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+      contents: read
+    outputs:
+      build-migrations: ${{ steps.filter.outputs.database }}
+    steps:
+    # For pull requests it's not necessary to checkout the code
+    - name: Checkout code
+      if: github.event_name != 'pull_request'
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - uses: dorny/paths-filter@v3
+      id: filter
+      with:
+        filters: |
+          database: &database
+            - 'src/database/PolicyHub.Migrations/**'
```

Comment:
I am not sure. I took reference from the upstream. I assumed it as necessary because there are 3 folders and only 2 are specified in the pipeline. 
Upstream ref: https://github.com/eclipse-tractusx/policy-hub/blob/2d93c25377809cd477d3c10c85717d015edbe9bc/.github/workflows/policy-hub-migrations.yml#L26